[Search] Use session service on a dashboard

Dosant opened the pull request:
## Release note

Adds Search Sessions support to Kibana Dashboards

## Summary

Follow up on https://github.com/elastic/kibana/pull/76889
Prev version of this pr with some comments: https://github.com/elastic/kibana/pull/81192
Part of https://github.com/elastic/kibana/issues/61738
Next step: restore searchSessionId from the URL: https://github.com/elastic/kibana/pull/81489

### Some context for reviewers

This is part of client side background search. To save or restore a background search, searches will be grouped under a session. We implemented session creation on every search in discover in #76889. This pr creates a search session on every dashboard container state change and propagates `searchSessionId` to embeddable -> search infrastructure. 

### Implementation notes:

1. Explicitly passes sessionId as embeddable input: 
  - [x] Visualize
  - [x] Lens
  - [x] Search
  - [ ] ~Maps: TODO: probably better done separately~ 
  - [ ] ~Anything else?~
2. For now this session id is only needed to group error toasts. For example, see two errors and a single toast. Before this pr you would see two separate toasts. The underlying code for grouping errors was added in https://github.com/elastic/kibana/pull/76889

![Screenshot 2020-10-21 at 17 12 13](https://user-images.githubusercontent.com/7784120/96739888-93d83200-13c0-11eb-8d0a-4f128f1e07b2.png)
3. Added `sessionId` displayed in inspector. Used it it in a functional test. I also think it would be useful for debug and support purposes. 
This is how it looks: 

![Screenshot 2020-10-21 at 17 07 54](https://user-images.githubusercontent.com/7784120/96739842-84f17f80-13c0-11eb-8d58-bc817fa21da0.png)

### Functional test:

Added a test that checks:

1. In case multiple panels failed, there is only single toast 
2. SessionId in inspector is the same for different panels
3. After state change and refetch sessionId changes

### How to test 

* You go to the dashboard and check that `searchSessionId` changes on different state changes. You can use inspector. 


### Open questions / edge cases to iterate over later
* Should refresh interval create a new session Id? - now it doesn't on a dashboard
* Internal embeddable state changes, like own filters, wouldn't create a new sessionId. This is fine.

### Next steps 

* Integrate`searchSessionId` with dashboardURL https://github.com/elastic/kibana/pull/81633 


### Checklist

Delete any items that are not applicable to this PR.

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83035/)
* Commit: 8f4b9aca53f0ac2728b8722e325cf761bb0d6f2b
* This comment will update when the build is complete

### Failed CI Steps
* [Check Doc API Changes](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83035/execution/node/375/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83035/","number":"83035","commit":"8f4b9aca53f0ac2728b8722e325cf761bb0d6f2b"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83041/)
* Commit: 45055e0d44d8eb80b820aedcd6c26ec10a209c18
* This comment will update when the build is complete

### Failed CI Steps
* [Check Doc API Changes](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83041/execution/node/372/log)

### History
* :broken_heart: [Build #83035](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83035/) failed 8f4b9aca53f0ac2728b8722e325cf761bb0d6f2b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83035/","number":"83035","commit":"8f4b9aca53f0ac2728b8722e325cf761bb0d6f2b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83041/","number":"83041","commit":"45055e0d44d8eb80b820aedcd6c26ec10a209c18"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83043/)
* Commit: 7c8660b87fb284862f6293b577340d39438548a6
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83043/execution/node/324/log)

### History
* :broken_heart: [Build #83041](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83041/) failed 45055e0d44d8eb80b820aedcd6c26ec10a209c18
* :broken_heart: [Build #83035](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83035/) failed 8f4b9aca53f0ac2728b8722e325cf761bb0d6f2b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83035/","number":"83035","commit":"8f4b9aca53f0ac2728b8722e325cf761bb0d6f2b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83041/","number":"83041","commit":"45055e0d44d8eb80b820aedcd6c26ec10a209c18"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83043/","number":"83043","commit":"7c8660b87fb284862f6293b577340d39438548a6"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83068/)
* Commit: 8fda2eefc6c7eb50652b85469ce796af7358a9d4
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83068/execution/node/364/log)

### History
* :broken_heart: [Build #83043](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83043/) failed 7c8660b87fb284862f6293b577340d39438548a6
* :broken_heart: [Build #83035](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83035/) failed 8f4b9aca53f0ac2728b8722e325cf761bb0d6f2b
* :broken_heart: [Build #83041](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83041/) failed 45055e0d44d8eb80b820aedcd6c26ec10a209c18

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83041/","number":"83041","commit":"45055e0d44d8eb80b820aedcd6c26ec10a209c18"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83035/","number":"83035","commit":"8f4b9aca53f0ac2728b8722e325cf761bb0d6f2b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83043/","number":"83043","commit":"7c8660b87fb284862f6293b577340d39438548a6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83068/","number":"83068","commit":"8fda2eefc6c7eb50652b85469ce796af7358a9d4"}]}
PIPELINE-->

Dosant commented on x-pack/test/functional/apps/dashboard/async_search/async_search.ts:
```diff
@@ -23,7 +26,7 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
     it('delayed should load', async () => {
       await PageObjects.common.navigateToApp('dashboard');
-      await PageObjects.dashboard.gotoDashboardEditMode('Delayed 5s');
+      await PageObjects.dashboard.loadSavedDashboard('Delayed 5s');
```
@lukasolson, I doubt that `gotoDashboardEditMode` was intentional, so I changed it to `loadSavedDashboard`

`loadSavedDashboard` is faster because we don't have to wait for initial load and then 2nd load after switch to edit mode.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83095/)
* Commit: 7fe2507018bcc9bad3163e7137b2411054328953
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83095/execution/node/378/log)

### History
* :broken_heart: [Build #83068](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83068/) failed 8fda2eefc6c7eb50652b85469ce796af7358a9d4
* :broken_heart: [Build #83043](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83043/) failed 7c8660b87fb284862f6293b577340d39438548a6
* :broken_heart: [Build #83035](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83035/) failed 8f4b9aca53f0ac2728b8722e325cf761bb0d6f2b
* :broken_heart: [Build #83041](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83041/) failed 45055e0d44d8eb80b820aedcd6c26ec10a209c18

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83041/","number":"83041","commit":"45055e0d44d8eb80b820aedcd6c26ec10a209c18"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83035/","number":"83035","commit":"8f4b9aca53f0ac2728b8722e325cf761bb0d6f2b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83043/","number":"83043","commit":"7c8660b87fb284862f6293b577340d39438548a6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83068/","number":"83068","commit":"8fda2eefc6c7eb50652b85469ce796af7358a9d4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83095/","number":"83095","commit":"7fe2507018bcc9bad3163e7137b2411054328953"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83103/)
* Commit: 7df03d1651473d70ed3263e051b95d170b65a8fe

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### async chunks size
| id | [before](https://github.com/elastic/kibana/commit/ad2fe78514825ad11ccaabec3a0ecd0977ad25eb) | [after](https://github.com/elastic/kibana/commit/7df03d1651473d70ed3263e051b95d170b65a8fe) | diff |
| --- | --- | --- | --- |
| `dashboard` | 221.2KB | 221.5KB | +324.0B |
| `discover` | 439.3KB | 439.6KB | +301.0B |
| `lens` | 1.1MB | 1.1MB | +252.0B |
| total |  |  | +877.0B |

#### page load bundle size
| id | [before](https://github.com/elastic/kibana/commit/ad2fe78514825ad11ccaabec3a0ecd0977ad25eb) | [after](https://github.com/elastic/kibana/commit/7df03d1651473d70ed3263e051b95d170b65a8fe) | diff |
| --- | --- | --- | --- |
| `dashboard` | 344.3KB | 344.4KB | +76.0B |
| `data` | 1.1MB | 1.1MB | +211.0B |
| `expressions` | 201.7KB | 202.0KB | +245.0B |
| `inspector` | 130.6KB | 131.2KB | +606.0B |
| `visualizations` | 272.2KB | 272.5KB | +262.0B |
| total |  |  | +1.4KB |

### History
* :broken_heart: [Build #83095](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83095/) failed 7fe2507018bcc9bad3163e7137b2411054328953
* :broken_heart: [Build #83068](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83068/) failed 8fda2eefc6c7eb50652b85469ce796af7358a9d4
* :broken_heart: [Build #83043](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83043/) failed 7c8660b87fb284862f6293b577340d39438548a6
* :broken_heart: [Build #83035](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83035/) failed 8f4b9aca53f0ac2728b8722e325cf761bb0d6f2b
* :broken_heart: [Build #83041](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83041/) failed 45055e0d44d8eb80b820aedcd6c26ec10a209c18

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83041/","number":"83041","commit":"45055e0d44d8eb80b820aedcd6c26ec10a209c18"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83035/","number":"83035","commit":"8f4b9aca53f0ac2728b8722e325cf761bb0d6f2b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83043/","number":"83043","commit":"7c8660b87fb284862f6293b577340d39438548a6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83068/","number":"83068","commit":"8fda2eefc6c7eb50652b85469ce796af7358a9d4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83095/","number":"83095","commit":"7fe2507018bcc9bad3163e7137b2411054328953"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83103/","number":"83103","commit":"7df03d1651473d70ed3263e051b95d170b65a8fe"}]}
PIPELINE-->

Dosant commented on src/plugins/dashboard/public/application/dashboard_app_controller.tsx:
```diff
@@ -1109,12 +1111,6 @@ export class DashboardAppController {
         $scope.model.query = queryStringManager.getQuery();
         dashboardStateManager.applyFilters($scope.model.query, $scope.model.filters);
-        if (dashboardContainer) {
-          dashboardContainer.updateInput({
```
This is not needed. After removing all dashboard container updates are handled in single place right now.
Looking forward to this file being refactored.

ThomThomson replied:
I'm in the middle of completely re-doing this file as part of deangularization. It will be a functional component - I may end up pinging you for some input at some point.

Dosant replied:
@ThomThomson, that's great news 👍 Will be glad to chime in. 
From perspective of this pr important bit is that we are calling `dashboardContainer.updateInput` in a single place

elasticmachine commented:
Pinging @elastic/kibana-app-arch (Team:AppArch)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83631/)
* Commit: 020f80964ea1c8ce79acd3dfaf84b6ea6d9918fc

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### async chunks size
| id | [before](https://github.com/elastic/kibana/commit/a8d8addacd15ee7523ec0c57e997ce4209c1d129) | [after](https://github.com/elastic/kibana/commit/020f80964ea1c8ce79acd3dfaf84b6ea6d9918fc) | diff |
| --- | --- | --- | --- |
| `dashboard` | 221.2KB | 221.5KB | +324.0B |
| `discover` | 439.1KB | 439.4KB | +301.0B |
| `lens` | 1.1MB | 1.1MB | +252.0B |
| total |  |  | +877.0B |

#### page load bundle size
| id | [before](https://github.com/elastic/kibana/commit/a8d8addacd15ee7523ec0c57e997ce4209c1d129) | [after](https://github.com/elastic/kibana/commit/020f80964ea1c8ce79acd3dfaf84b6ea6d9918fc) | diff |
| --- | --- | --- | --- |
| `dashboard` | 344.3KB | 344.4KB | +76.0B |
| `data` | 1.1MB | 1.1MB | +211.0B |
| `inspector` | 130.6KB | 131.2KB | +606.0B |
| `visualizations` | 272.2KB | 272.5KB | +262.0B |
| total |  |  | +1.1KB |

### History
* :green_heart: [Build #83103](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83103/) succeeded 7df03d1651473d70ed3263e051b95d170b65a8fe
* :broken_heart: [Build #83095](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83095/) failed 7fe2507018bcc9bad3163e7137b2411054328953
* :broken_heart: [Build #83068](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83068/) failed 8fda2eefc6c7eb50652b85469ce796af7358a9d4
* :broken_heart: [Build #83043](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83043/) failed 7c8660b87fb284862f6293b577340d39438548a6
* :broken_heart: [Build #83035](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83035/) failed 8f4b9aca53f0ac2728b8722e325cf761bb0d6f2b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83035/","number":"83035","commit":"8f4b9aca53f0ac2728b8722e325cf761bb0d6f2b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83043/","number":"83043","commit":"7c8660b87fb284862f6293b577340d39438548a6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83068/","number":"83068","commit":"8fda2eefc6c7eb50652b85469ce796af7358a9d4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83095/","number":"83095","commit":"7fe2507018bcc9bad3163e7137b2411054328953"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83103/","number":"83103","commit":"7df03d1651473d70ed3263e051b95d170b65a8fe"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83631/","number":"83631","commit":"020f80964ea1c8ce79acd3dfaf84b6ea6d9918fc"}]}
PIPELINE-->

lukasolson commented on src/plugins/inspector/public/views/requests/components/requests_view.tsx:
```diff
@@ -153,6 +153,21 @@ export class RequestsViewComponent extends Component<InspectorViewProps, Request
           </EuiText>
         )}
 
+        {this.state.request && this.state.request.searchSessionId && (
```
Is this only for the functional tests, or do you think there is value in showing this parameter to the user?

Dosant replied:
I think it would also be useful for debugging and support purposes

lukasolson reviewed (approved):
LGTM, verified that the `searchSessionId` seems to be set properly across visualizations that use `SearchSource`. It's updated properly when the filter/query/time range is updated and a new query is sent out.

ThomThomson reviewed (approved):
Code & behaviours LGTM. 

Tested locally in chrome by making changes in the dashboard app and ensuring that every embeddable uses the same `sessionId`, and that the `sessionId` changes correctly with every change.

Dosant commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84227/)
* Commit: fda39c4bd7f50464cb6034f60145b295819a301d

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### async chunks size
| id | [before](https://github.com/elastic/kibana/commit/e160d549701bcff404f10fe7c8085af50707319e) | [after](https://github.com/elastic/kibana/commit/fda39c4bd7f50464cb6034f60145b295819a301d) | diff |
| --- | --- | --- | --- |
| `dashboard` | 221.2KB | 221.5KB | +324.0B |
| `discover` | 439.6KB | 439.9KB | +301.0B |
| `lens` | 1.1MB | 1.1MB | +252.0B |
| total |  |  | +877.0B |

#### page load bundle size
| id | [before](https://github.com/elastic/kibana/commit/e160d549701bcff404f10fe7c8085af50707319e) | [after](https://github.com/elastic/kibana/commit/fda39c4bd7f50464cb6034f60145b295819a301d) | diff |
| --- | --- | --- | --- |
| `dashboard` | 344.4KB | 344.4KB | +76.0B |
| `data` | 1.1MB | 1.1MB | +211.0B |
| `inspector` | 130.6KB | 131.2KB | +606.0B |
| `visualizations` | 272.2KB | 272.5KB | +262.0B |
| total |  |  | +1.1KB |

### History
* :green_heart: [Build #83631](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83631/) succeeded 020f80964ea1c8ce79acd3dfaf84b6ea6d9918fc
* :green_heart: [Build #83103](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83103/) succeeded 7df03d1651473d70ed3263e051b95d170b65a8fe
* :broken_heart: [Build #83095](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83095/) failed 7fe2507018bcc9bad3163e7137b2411054328953
* :broken_heart: [Build #83068](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83068/) failed 8fda2eefc6c7eb50652b85469ce796af7358a9d4
* :broken_heart: [Build #83043](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83043/) failed 7c8660b87fb284862f6293b577340d39438548a6

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83043/","number":"83043","commit":"7c8660b87fb284862f6293b577340d39438548a6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83068/","number":"83068","commit":"8fda2eefc6c7eb50652b85469ce796af7358a9d4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83095/","number":"83095","commit":"7fe2507018bcc9bad3163e7137b2411054328953"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83103/","number":"83103","commit":"7df03d1651473d70ed3263e051b95d170b65a8fe"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83631/","number":"83631","commit":"020f80964ea1c8ce79acd3dfaf84b6ea6d9918fc"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84227/","number":"84227","commit":"fda39c4bd7f50464cb6034f60145b295819a301d"}]}
PIPELINE-->

flash1293 commented:
There's something weird going on I think. If I hit the refresh button, the `Aborted` label flashes in saved searches:
![abortedflash](https://user-images.githubusercontent.com/1508364/97545731-b0c9c200-19cb-11eb-94f3-b8ccf80d90f9.gif)

I debugged into the Lens embeddable and for a single hit of refresh, it's calling `render` twice - once with the old session id, then with the new session id (it's probably the same for the saved search, that's why the label flashes). On master it's just called once.

lizozom commented on docs/development/plugins/embeddable/public/kibana-plugin-plugins-embeddable-public.embeddableinput.md:
```diff
@@ -19,5 +19,6 @@ export declare type EmbeddableInput = {
     timeRange?: TimeRange;
     query?: Query;
     filters?: Filter[];
+    searchSessionId?: string;
```
This is a major nit, but do we really want to have the name be so specific?
I think sessions can and will be used for more than search.

Dosant replied:
I think `sessionId` is way too vague for dashboards / embeddable level, but OK on search level. 
`sessionId` on dashboard could mean anything: 
* user authentication session 
* dashboard page session (created once user when user loaded a dashboard) 
* dashboard app session (to track drill-downs between dashboards without navigating away from a dashboard app)
* and the search / fetch session.

lizozom replied:
Agreed, if this terminology is for using in solutions :)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84494/)
* Commit: 898686749a35e48ad10f80543fcf7c2a67e9fd26
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84494/execution/node/321/log)

### History
* :green_heart: [Build #84227](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84227/) succeeded fda39c4bd7f50464cb6034f60145b295819a301d
* :green_heart: [Build #83631](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83631/) succeeded 020f80964ea1c8ce79acd3dfaf84b6ea6d9918fc
* :green_heart: [Build #83103](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83103/) succeeded 7df03d1651473d70ed3263e051b95d170b65a8fe
* :broken_heart: [Build #83095](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83095/) failed 7fe2507018bcc9bad3163e7137b2411054328953
* :broken_heart: [Build #83068](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83068/) failed 8fda2eefc6c7eb50652b85469ce796af7358a9d4

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83068/","number":"83068","commit":"8fda2eefc6c7eb50652b85469ce796af7358a9d4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83095/","number":"83095","commit":"7fe2507018bcc9bad3163e7137b2411054328953"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83103/","number":"83103","commit":"7df03d1651473d70ed3263e051b95d170b65a8fe"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83631/","number":"83631","commit":"020f80964ea1c8ce79acd3dfaf84b6ea6d9918fc"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84227/","number":"84227","commit":"fda39c4bd7f50464cb6034f60145b295819a301d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84494/","number":"84494","commit":"898686749a35e48ad10f80543fcf7c2a67e9fd26"}]}
PIPELINE-->

Dosant commented:
@flash1293,

> There's something weird going on I think. If I hit the refresh button, the `Aborted` label flashes in saved searches:
> ![abortedflash](https://user-images.githubusercontent.com/1508364/97545731-b0c9c200-19cb-11eb-94f3-b8ccf80d90f9.gif)
> 
> I debugged into the Lens embeddable and for a single hit of refresh, it's calling `render` twice - once with the old session id, then with the new session id (it's probably the same for the saved search, that's why the label flashes). On master it's just called once.

Thanks for pointing that out. There were two issue I found and fixed, let's see what tests say...

1. when embeddable was updated with input: `{lastRequestTime, searchSessionId}` first it was reload with OLDER input. and then input was updated with `searchSessionId`. SearchSessionId change was triggering the render again. I changed the order, so now `reload` is called after input is update.
2. Initially I made embeddables to update when `searchSessionId` changes, but this was causing issue that in case of input update like: `{lastRequestTime, searchSessionId}` we trigger two renders if embeddable is not handling `lastRequestTime` comparison internally. Lens does, but other don't. I decided to revert `searchSessionId` checking in internal embeddable state change detection and just forward `this.input.searchSessionId` during rendering assuming that `searchSessionId` can never change by itself (which seems to be the case)

lizozom commented on src/plugins/visualizations/public/embeddable/visualize_embeddable.ts:
```diff
@@ -374,6 +374,7 @@ export class VisualizeEmbeddable
         query: this.input.query,
         filters: this.input.filters,
       },
+      searchSessionId: this.input.searchSessionId,
```
If we pass in the `searchSessionId` from the visualize app, would that enable sessions for that app too?

lizozom commented on x-pack/test/functional/apps/dashboard/async_search/async_search.ts:
```diff
@@ -42,10 +45,47 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
+        return searchSessionId;
+      };
+
+      const panel1SessionId1 = await getSearchSessionIdByPanel('Sum of Bytes by Extension');
```
Would it make sense to add tests to `test/plugin_functional/test_suites/data_plugin/session.ts` to make sure that sessions are being created once per load \ reload?

It was what uncovered the problematic behaviors in Dashboard in the first place.

Dosant replied:
I will take a look 👍 
but would like to do it in separate pr, 
really want to unblock a bunch of follow up prs with merging this :(

lizozom reviewed (commented):
Looks good.
Added a couple of suggestions

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84499/)
* Commit: 368cedf017a6f391c02967d15b4313816af94234

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### @kbn/ui-shared-deps asset size
| id | [before](https://github.com/elastic/kibana/commit/06c99bbf7e102683414809c99cffa636af54eaaf) | [after](https://github.com/elastic/kibana/commit/368cedf017a6f391c02967d15b4313816af94234) | diff |
| --- | --- | --- | --- |
| `css` | - | 645.3KB | :warning: +645.3KB |
| `kbn-ui-shared-deps.@elastic.js` | - | 2.4MB | :warning: +2.4MB |
| `kbn-ui-shared-deps.js` | - | 4.8MB | :warning: +4.8MB |
| total |  |  | :warning: +7.8MB |

#### async chunks size
| id | [before](https://github.com/elastic/kibana/commit/06c99bbf7e102683414809c99cffa636af54eaaf) | [after](https://github.com/elastic/kibana/commit/368cedf017a6f391c02967d15b4313816af94234) | diff |
| --- | --- | --- | --- |
| `dashboard` | 193.8KB | 194.1KB | +296.0B |
| `discover` | 407.0KB | 407.1KB | +118.0B |
| `lens` | 1.0MB | 1.0MB | +107.0B |
| total |  |  | +521.0B |

#### page load bundle size
| id | [before](https://github.com/elastic/kibana/commit/06c99bbf7e102683414809c99cffa636af54eaaf) | [after](https://github.com/elastic/kibana/commit/368cedf017a6f391c02967d15b4313816af94234) | diff |
| --- | --- | --- | --- |
| `dashboard` | 310.2KB | 310.3KB | +76.0B |
| `data` | 944.9KB | 945.1KB | +165.0B |
| `embeddable` | 217.5KB | 217.6KB | +57.0B |
| `inspector` | 83.0KB | 83.6KB | +606.0B |
| `visualizations` | 207.2KB | 207.2KB | +45.0B |
| total |  |  | +949.0B |

### History
* :broken_heart: [Build #84494](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84494/) failed 898686749a35e48ad10f80543fcf7c2a67e9fd26
* :green_heart: [Build #84227](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84227/) succeeded fda39c4bd7f50464cb6034f60145b295819a301d
* :green_heart: [Build #83631](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83631/) succeeded 020f80964ea1c8ce79acd3dfaf84b6ea6d9918fc
* :green_heart: [Build #83103](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83103/) succeeded 7df03d1651473d70ed3263e051b95d170b65a8fe
* :broken_heart: [Build #83095](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83095/) failed 7fe2507018bcc9bad3163e7137b2411054328953

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83095/","number":"83095","commit":"7fe2507018bcc9bad3163e7137b2411054328953"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83103/","number":"83103","commit":"7df03d1651473d70ed3263e051b95d170b65a8fe"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83631/","number":"83631","commit":"020f80964ea1c8ce79acd3dfaf84b6ea6d9918fc"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84227/","number":"84227","commit":"fda39c4bd7f50464cb6034f60145b295819a301d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84494/","number":"84494","commit":"898686749a35e48ad10f80543fcf7c2a67e9fd26"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84499/","number":"84499","commit":"368cedf017a6f391c02967d15b4313816af94234"}]}
PIPELINE-->

Dosant commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84528/)
* Commit: 0bbeb44d7a5b86c848069334e7ccff2cabe87d68

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### async chunks size
| id | [before](https://github.com/elastic/kibana/commit/275c30a926a3bde5836180b726389d73de39addf) | [after](https://github.com/elastic/kibana/commit/0bbeb44d7a5b86c848069334e7ccff2cabe87d68) | diff |
| --- | --- | --- | --- |
| `dashboard` | 193.8KB | 194.1KB | +296.0B |
| `discover` | 407.0KB | 407.1KB | +118.0B |
| `lens` | 1.0MB | 1.0MB | +107.0B |
| total |  |  | +521.0B |

#### page load bundle size
| id | [before](https://github.com/elastic/kibana/commit/275c30a926a3bde5836180b726389d73de39addf) | [after](https://github.com/elastic/kibana/commit/0bbeb44d7a5b86c848069334e7ccff2cabe87d68) | diff |
| --- | --- | --- | --- |
| `dashboard` | 310.2KB | 310.3KB | +76.0B |
| `data` | 944.9KB | 945.1KB | +165.0B |
| `embeddable` | 217.5KB | 217.6KB | +57.0B |
| `inspector` | 83.0KB | 83.6KB | +606.0B |
| `visualizations` | 207.1KB | 207.2KB | +45.0B |
| total |  |  | +949.0B |

### History
* :green_heart: [Build #84499](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84499/) succeeded 368cedf017a6f391c02967d15b4313816af94234
* :broken_heart: [Build #84494](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84494/) failed 898686749a35e48ad10f80543fcf7c2a67e9fd26
* :green_heart: [Build #84227](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84227/) succeeded fda39c4bd7f50464cb6034f60145b295819a301d
* :green_heart: [Build #83631](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83631/) succeeded 020f80964ea1c8ce79acd3dfaf84b6ea6d9918fc
* :green_heart: [Build #83103](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83103/) succeeded 7df03d1651473d70ed3263e051b95d170b65a8fe

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83103/","number":"83103","commit":"7df03d1651473d70ed3263e051b95d170b65a8fe"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83631/","number":"83631","commit":"020f80964ea1c8ce79acd3dfaf84b6ea6d9918fc"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84227/","number":"84227","commit":"fda39c4bd7f50464cb6034f60145b295819a301d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84494/","number":"84494","commit":"898686749a35e48ad10f80543fcf7c2a67e9fd26"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84499/","number":"84499","commit":"368cedf017a6f391c02967d15b4313816af94234"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84528/","number":"84528","commit":"0bbeb44d7a5b86c848069334e7ccff2cabe87d68"}]}
PIPELINE-->

flash1293 reviewed (approved):
Lens and saved searches LGTM - it seems like the search session id is passed through correctly. It seems like Discover is fetching twice on changing the time range, but that happens on master as well. I will open a separate issue for this.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/108513/)
* Commit: 0bbeb44d7a5b86c848069334e7ccff2cabe87d68
* This comment will update when the build is complete

### Failed CI Steps
* [Verify NOTICE](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/108513/execution/node/395/log)

### History
* :green_heart: [Build #84528](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84528/) succeeded 0bbeb44d7a5b86c848069334e7ccff2cabe87d68
* :green_heart: [Build #84499](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84499/) succeeded 368cedf017a6f391c02967d15b4313816af94234
* :broken_heart: [Build #84494](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84494/) failed 898686749a35e48ad10f80543fcf7c2a67e9fd26
* :green_heart: [Build #84227](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84227/) succeeded fda39c4bd7f50464cb6034f60145b295819a301d
* :green_heart: [Build #83631](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83631/) succeeded 020f80964ea1c8ce79acd3dfaf84b6ea6d9918fc

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/83631/","number":"83631","commit":"020f80964ea1c8ce79acd3dfaf84b6ea6d9918fc"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84227/","number":"84227","commit":"fda39c4bd7f50464cb6034f60145b295819a301d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84494/","number":"84494","commit":"898686749a35e48ad10f80543fcf7c2a67e9fd26"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84499/","number":"84499","commit":"368cedf017a6f391c02967d15b4313816af94234"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/84528/","number":"84528","commit":"0bbeb44d7a5b86c848069334e7ccff2cabe87d68"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/108513/","number":"108513","commit":"0bbeb44d7a5b86c848069334e7ccff2cabe87d68"}]}
PIPELINE-->
